crates/nu-cli/src/shell/completer.rs

```diff
@@ -19,12 +26,32 @@ impl NuCompleter {
             Ok(block) => Some(block),
             Err(result) => result.partial,
         };
-
+                                    
         let locations = lite_block
             .map(|block| nu_parser::classify_block(&block, &nu_context.registry))
             .map(|block| crate::completion::engine::completion_location(line, &block.block, pos))
             .unwrap_or_default();
 
+        let config = config_or_empty(Tag::unknown());
+        let matcher_config_optional: Option<String> = config.get("completion.matcher")
+            .and_then(|value| match value.as_string() {
+                Ok(result) => Some(result),
+                Err(_) => Some(String::from(""))
+            });
+
+        let matcher_config: String = match matcher_config_optional {
+            Some(value) => value,
+            None => String::from("")
+        };
+    
+        let matcher_config: &str = matcher_config.as_str();
```

Comment:
this works, but seems verbose.

Reply:
Verbose in what way?

Reply:
I struggled hard with getting that config value out with default values, wondering if there is a more idiomatic way,

i.e. ruby-esque:

matcher_config = config.get('completion.matcher') || 'case-insensitive'